src/components/Icon/Icon.scss

```diff
@@ -27,192 +27,394 @@
   }
 }
 
 .colorGreenDark {
   @include color-icon('green', 'dark');
 
   &::before {
-    background-color: color('green', 'light');
+    background-color: var(--p-surface-success, color('green', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-success));
   }
 }
 
 .colorYellowLighter {
   @include color-icon('yellow', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorYellow {
   @include color-icon('yellow');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorYellowDark {
   @include color-icon('yellow', 'dark');
 
   &::before {
-    background-color: color('yellow', 'light');
+    background-color: var(--p-surface-warning, color('yellow', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
   }
 }
 
 .colorOrange {
   @include color-icon('orange');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorOrangeDark {
   @include color-icon('orange', 'dark');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-warning));
+  }
 }
 
 .colorRedLighter {
   @include color-icon('red', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-critical));
+  }
 }
 
 .colorRed {
   @include color-icon('red');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-critical));
+  }
 }
 
 .colorRedDark {
   @include color-icon('red', 'dark');
 
   &::before {
-    background-color: color('red', 'light');
+    background-color: var(--p-surface-critical, color('red', 'light'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon-critical));
   }
 }
 
 .colorPurple {
   @include color-icon('purple');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
+}
+
+// New design language color styles
+
+.colorBase {
+  @include recolor-icon(var(--p-icon));
+}
+
+.colorDisabled {
+  @include recolor-icon(var(--p-icon-disabled));
+}
+
+.colorSubdued {
+  @include recolor-icon(var(--p-icon-subdued));
+}
+
+.colorHovered {
+  @include recolor-icon(var(--p-icon-hovered));
+}
+
+.colorPressed {
+  @include recolor-icon(var(--p-icon-pressed));
+}
+
+.colorCritical {
+  @include recolor-icon(var(--p-icon-critical));
+}
+
+.colorWarning {
+  @include recolor-icon(var(--p-icon-warning));
+}
+
+.colorHighlight {
+  @include recolor-icon(var(--p-icon-highlight));
+}
+
+.colorSuccess {
+  @include recolor-icon(var(--p-icon-success));
+}
+
+.colorInteractive {
+  @include recolor-icon(var(--p-interactive));
+}
+
+.colorPrimary {
+  @include recolor-icon(var(--p-action-primary));
+}
+
+.colorPrimaryDisabled {
+  @include recolor-icon(var(--p-action-primary-disabled));
+}
+
+.colorPrimaryHovered {
+  @include recolor-icon(var(--p-action-primary-hovered));
+}
+
+.colorPrimaryPressed {
+  @include recolor-icon(var(--p-action-primary-pressed));
 }
```

Comment:
These are the style for all the icon colors added to support the new design language.